tools/python/src/image2.cpp

```diff
@@ -28,6 +28,20 @@ numpy_image<T> py_resize_image (
 
 // ----------------------------------------------------------------------------------------
 
+template <typename T>
+numpy_image<T> py_scale_image (
+    const numpy_image<T>& img,
+    double scale
+)
+{
+    DLIB_CASSERT(scale > 0, "Scale factor must be greater than 0");
+    numpy_image<T> out = img;
```

Comment:
I'm pretty sure this will result in the input image being modified.  Isn't that what happens?

You need to call the version of resize_image() that takes two images, an in and an out.

Reply:
When I test in python it works as expected and is not modifying the input.

As I understand it and my intention here is that `out` is a copy of `img` instantiated via the implicit copy constructor of `numpy_image`. The copy `out` is modified when passed to `scale_image` and returned without modifying the original input `img`.

I have little experience with c++ so was hoping to implement the functionality whilst touching as few lines as possible - for good or bad. Happy to give it another shot if you would still prefer it implemented as suggested.